ide-backend/IdeSession/Update/ExecuteSessionUpdate.hs

```diff
@@ -579,37 +587,40 @@ markAsUpdated shouldMark = do
 -- TODO: We should verify each use of exceptionFree here.
 executeFileCmd :: FileCmd -> ExecuteSessionUpdate Bool
 executeFileCmd cmd = do
-  IdeStaticInfo{..} <- asks ideUpdateStaticInfo
+  ideStaticInfo@IdeStaticInfo{..} <- asks ideUpdateStaticInfo
 
   let remotePath :: FilePath
-      remotePath = fileInfoRemoteDir info ideSessionDir </> fileInfoRemoteFile info
-
-  case cmd of
-    FileWrite _ bs -> do
-      old <- Acc.get cachedInfo
-      -- We always overwrite the file, and then later set the timestamp back
-      -- to what it was if it turns out the hash was the same. If we compute
-      -- the hash first, we would force the entire lazy bytestring into memory
-      newHash <- exceptionFree $ writeFileAtomic remotePath bs
-      case old of
-        Just (oldHash, oldTS) | oldHash == newHash -> do
-          exceptionFree $ setFileTimes remotePath oldTS oldTS
-          return False
-        _ -> do
-          newTS <- updateFileTimes remotePath
-          Acc.set cachedInfo (Just (newHash, newTS))
-          return True
-    FileCopy _ localFile -> do
-      -- We just call 'FileWrite' because we need to read the file anyway to
-      -- compute the hash. Note that `localPath` is interpreted relative to the
-      -- current directory
-      bs <- exceptionFree $ BSL.readFile localFile
-      executeFileCmd (FileWrite info bs)
-    FileDelete _ -> do
-      exceptionFree $ ignoreDoesNotExist $ Dir.removeFile remotePath
-      Acc.set cachedInfo Nothing
-      -- TODO: We should really return True only if the file existed
-      return True
+      remotePath = fileInfoRemoteDir info ideStaticInfo </> fileInfoRemoteFile info
+  
+  case configLocalWorkingDir ideConfig of
```

Comment:
Cool!  I like putting the check here, I'd imagined needing to change all the update functions.  This is better.